src/easylink/step.py

```diff
@@ -99,9 +110,14 @@ def update_implementation_graph(
     def update_edges(self, graph: nx.MultiDiGraph, step_config: LayeredConfigTree) -> None:
         """Add edges to/from the implementation node to replace the edges from the current step"""
         implementation_name = step_config["implementation"]["name"]
+        implementation_node_name = (
+            implementation_name
+            if self.name == self.step_name
+            else f"{self.name}_{implementation_name}"
+        )
```

Comment:
Would it make sense to factor this out? I see it is also above.

Reply:
i actually do extract this in the next PR